lib/src/protocols/websocket_io.dart

```diff
@@ -65,6 +65,7 @@ class KuzzleWebSocket extends KuzzleProtocol {
         onError: _handleError, onDone: _handleDone);
 
     clientConnected();
+    _stopRetryingToConnect = false;
```

Comment:
I think there is one more edge case, here you reset it back when the connection is successful, meaning that, in the case of [`current connection` -> `close()` -> `connection fails`], the variable `_stopRetryingToConnect` will be `false` preventing the auto reconnection attempts

Reply:
If you close the connection then it does not make sense to have the reconnection attempts. To have reconnection attempts you first need to be connected (_stopRetryingToConnect will then be false), have a disconnection and then it will start trying  to reconnect.

Reply:
I've searched across our different SDKs, and the current behaviour is if a connection fails and the autoReconnect is `true` start a reconnection attempt, even if that was the first connection.

This is even the case here in the dark  SDK
```dart
if (wasConnected || autoReconnect) {
        clientNetworkError(
            KuzzleError('WebSocketProtocol: Unable to connect to $url'));
        _handleAutoReconnect();
}
````